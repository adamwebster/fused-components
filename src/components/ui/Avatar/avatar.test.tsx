import React from 'react';
import { render, cleanup } from '@testing-library/react';
import 'jest-styled-components';
import { Avatar } from './index';

afterEach(cleanup);

describe('Avatar Tests', () => {
  test('Renders the component', () => {
    render(<Avatar />);
  });
  test('Changing the size to medium has the correct size', () => {
    const { container } = render(<Avatar size="medium" />);
    expect(container).toMatchSnapshot();
  });
  test('Changing the size to tiny has the correct size', () => {
    const { container } = render(<Avatar size="tiny" />);
    expect(container).toMatchSnapshot();
  });
  test('Changing the size to large has the correct size', () => {
    const { container } = render(<Avatar size="large" />);
    expect(container).toMatchSnapshot();
  });
  test('Adding the box shadow prop matches the snapshot', () => {
    const { container } = render(<Avatar boxShadow />);
    expect(container).toMatchSnapshot();
  });
  test('Setting the border radius to square matches the snapshot', () => {
    const { container } = render(<Avatar borderRadius="square" />);
    expect(container).toMatchSnapshot();
  });
  test('Setting the border radius to round matches the snapshot', () => {
    const { container } = render(<Avatar borderRadius="round" />);
    expect(container).toMatchSnapshot();
  });
  test('Adding an image passes the url to the css', () => {
    const { container } = render(<Avatar image="http://localhost/TestImage.jpg" />);
    expect(container).toMatchSnapshot();
  });
});
