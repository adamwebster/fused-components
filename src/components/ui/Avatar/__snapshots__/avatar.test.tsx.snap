// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Tests Adding an image passes the url to the css 1`] = `
.c0 {
  width: 24px;
  height: 24px;
  border-radius: 10%;
  background-image: url(http://localhost/TestImage.jpg);
  background-size: cover;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Avatar Tests Adding the box shadow prop matches the snapshot 1`] = `
.c0 {
  width: 24px;
  height: 24px;
  border-radius: 10%;
  background-image: url();
  background-size: cover;
  box-shadow: 0 0 5px rgba(0,0,0,0.4);
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Avatar Tests Changing the size to large has the correct size 1`] = `
.c0 {
  width: 64px;
  height: 64px;
  border-radius: 10%;
  background-image: url();
  background-size: cover;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Avatar Tests Changing the size to medium has the correct size 1`] = `
.c0 {
  width: 44px;
  height: 44px;
  border-radius: 10%;
  background-image: url();
  background-size: cover;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Avatar Tests Changing the size to tiny has the correct size 1`] = `
.c0 {
  width: 16px;
  height: 16px;
  border-radius: 10%;
  background-image: url();
  background-size: cover;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Avatar Tests Setting the border radius to round matches the snapshot 1`] = `
.c0 {
  width: 24px;
  height: 24px;
  border-radius: 50%;
  background-image: url();
  background-size: cover;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Avatar Tests Setting the border radius to square matches the snapshot 1`] = `
.c0 {
  width: 24px;
  height: 24px;
  border-radius: 0;
  background-image: url();
  background-size: cover;
}

<div>
  <div
    class="c0"
  />
</div>
`;
